pandas/core/frame.py

```diff
@@ -6073,7 +6073,7 @@ def notnull(self) -> DataFrame:
     def dropna(
         self,
         axis: Axis = 0,
-        how: str = "any",
+        how: str | None = None,
```

Comment:
if users aren't supposed to pass `None`, this should probably be `lib.no_default`, e.g.

https://github.com/pandas-dev/pandas/blob/361021b56f3159afb71d690fac3a1f3b381b0da6/pandas/core/generic.py#L7935

Reply:
Are you sure `no_default` is appropriate? Most of its usage if for a method parameter that is deprecated. So if this parameter `is not no_default` a warning could be raised. This is not the case here. Users aren't supposed to pass anything but "any" or "all", and the code correctly raises if how is not "any" or "all". 

Using `no_default` seems semantically wrong. If you confirm `how` must default to `no_default` I am fine with it, but I want to cross check with you.

Reply:
Let's check with @simonjayhawkins

Reply:
I think the following use cases cover lib.no_default

- when we need to detect that a user explicitly passed `None` where `None` is a valid value, i.e. hashable index label
- where there is a non `None` default and we need to detect if a user passed any value if the keyword arg is depecated so that we can raise a warning.
- where users must not specify two or more kwargs arguments simultaneously and we need to detect that the user passed values to distinguish from the defaults (for validation)

so at first glance it appears this issues falls into the third category.

> Users aren't supposed to pass anything but "any" or "all", and the code correctly raises if `how` is not "any" or "all".

so it really depends on the order of validation on whether we need to use lib.no_default.

I would have though that the exception for explicit passing of both `how` and `thresh` regardless of the values passed should take precedence over the exception for passing anything but "any" or "all" to `how`?

so the question is should `df.dropna(how=None, thresh=2)` raise? With this PR it gives an empty DataFrame (the same as 1.4.2). But also `df.dropna(how=object(), thresh=2)` returns an empty DataFrame for 1.4.2 and now correctly raises `TypeError: You cannot set both the how and thresh arguments at the same time.`

Also, `df.dropna(how=None)` returns the same as `df.dropna(how="any")` whereas on 1.4.2 this gave `TypeError: must specify how or thresh`. This was tested behavior and the test for this has been removed in this PR.